readme.md

```diff
@@ -30,133 +30,148 @@ Then, go to [http://localhost:5000/](http://localhost:5000) and you should see a
 
 * **Question:** Why do we get a response when we go to `/` but not `/notfound`?
 
 * **Your Answer:**
 
----
+---/notfound enpoint doesnt exist - so the we get the error
+get / returns the response from app.js - which is the message dsiplayed
 
 - [ ] By default, browsers make `GET` requests when you go to a URL. Use [Postman](https://www.getpostman.com) to make a similar `GET` request.
 
----
+---done
 
 - [ ] In Postman, go to the **Headers** tab _that is on the same level as the response Body._ These are headers that were returned by your server.
 
 * **Question:** What are headers? What values do you recognize?
 
 * **Your Answer:**
 
----
+---X-Powered by - tells on which application was built - here Express
+Content type and length - represents type and length of message body in the response
 
 - [ ] Add the following line **above** the `res.json()` line in your `app.js` file; then, make the request again.
   ```js
-  console.log(req.headers)
+  console.log(req.headers);
   ```
 
 * **Question:** Where can you find this `console.log()` statement? How can you change the headers that are sent in Postman?
 
 * **Your Answer:**
 
----
+---console.log() is displayed on the terminal.
+parameters could be sent through req.headers object -> changes the headers in postman
 
 - [ ] Add the following line **above** the `res.json()` line in your `app.js` file; then, make a request again but this time change the URL to the following: `http://localhost:5000?course=javascript`
   ```js
-  console.log(req.query)
+  console.log(req.query);
   ```
 
 * **Question:** What are query parameters? Try going to a couple of your favorite websites and paste an example of query parameters being used.
 
 * **Your Answer:**
 
----
+---{ course: 'javascript' } ->query parameters
+
+Query parameters are optional key-value pairs that appear to the right of the ? in a URL.
+
+Example:
+?module_item_id=9525216
 
 - [ ] Before all of your routes, add the following:
   ```js
   app.use((req, res, next) => {
-    console.log('In the server!')
-    next()
-  })
+    console.log("In the server!");
+    next();
+  });
   ```
 
 * **Question:** When does `app.use()` get called?
 
 * **Your Answer:**
 
----
+---Everytime a request is sent to the server app.use is called
 
 - [ ] Take a moment to observe the basic structure of a **route** in Express.
   ```js
-  app.get('/', (req, res, next) => {
-    console.log(req.query)
+  app.get("/", (req, res, next) => {
+    console.log(req.query);
     res.json({
-      message: 'Hello, Express!'
-    })
-  })
+      message: "Hello, Express!"
+    });
+  });
   ```
 
 * **Question:** What type of thing is `app` and what is its purpose?
 
-* **Your Answer:**
+* **Your Answer:**a function - instance of express on which methods are invoked
 
 * **Question:** What type of thing is `app.get()` and what is its purpose?
 
-* **Your Answer:**
+* **Your Answer:**finds a route for the GET request
```

Comment:
This is not quite correct. `app.get()` is a function and it sets up a new route. I'm not sure if "find" is the right term here.